File: Work2/3.cpp
#include <iostream>
#include <cmath>

using namespace std;

double calculate_exp(double x, double constant = 0.0001) {
    double result = 0;
    double term = x;
    int n = 1;

    while (abs(term) >= constant) {
        result += term;
        n++;
        term = pow(x, n) / tgamma(n + 1);
    }

    return result;
}

int main() {
    double x;

    cout << "Input x: ";
    cin >> x;
    double exp = calculate_exp(x);

    cout << "Exp: " << exp << endl;

    return 0;
}

File: Work2/3.py
import math
x = float(input("Input x"))
def calculate_exp(x, constant=0.0001):
    result = 0
    term = x
    n = 1

    while abs(term) >= constant:
        result += term
        n += 1
        term = x**n / math.factorial(n)

    return result
exp = calculate_exp(x)
print("Exp: ", exp)
